fabric-network/src/impl/wallet/hsmx509identity.ts

```diff
@@ -65,11 +67,12 @@ export class HsmX509Provider implements IdentityProvider {
 			throw new Error('Invalid identity type: ' + data.type);
 		}
 
-		if (data.version === 1) {
+		if (data.version === 2) {
```

Comment:
There still needs to be an `if (data.version === 1)` block to handle existing (v1) HSM identities stored in wallets. It should either:

1. Create an `HsmX509Identity` object directly from the v1 data and return it; or
1. Convert the v1 format data to v2 format data and then let it fall through to the `if (data.version === 2)` block.

Either way, there should be an `if (data.version === 2)` block in addition to handle the v2 shape data directly.